ni/niveristand-communications-bus-template
# Cloning - final architecture and VIs to build a working comms clone.

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-communications-bus-template/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

The cloning process is complete, and creates a working communications custom device. Projects, PPLs, LLBs, LVPROJ, XML, and CHM files all get renamed/updated/rebuilt as needed to create a unique clone. The Cloning UI includes a progress bar, build time, and Cancel button.

Edited VIs:

- `Add Port RTM.vi` was edited to fix a bug - some culprit accidentally saved this with all the controls scrolled off screen.
- `Build CHM.vi` - added an error indicator, so we can ignore the error. Otherwise this VI opens during our cloning precess.
- `Clone Template Main.vi` - added an architecture with a UI for customers.
- Most of the remaining edited VIs were due to the new architecture where I was able to remove `Template Tool Global.vi` and use a data cluster instead.

New Files:

- `Consumer Data.ctl`, `Consumer State.ctl`, `Queue Cluster.ctl`, and `Milliseconds to Minutes Seconds.vi` were added for the UI architecture.
- **Delete** VIs are used to delete files or folders.
- **Edit** VIs basically change an item in a file.
- **Rename** VIs either rename/replace items in a project or rename items directly in a text file.
- **Update and Build** VIs update build specs to recover lost information from renaming and then perform a build.

### Why should this Pull Request be merged?

Except for documentation, this finishes the **Simpler method for cloning the Comms Bus Template** feature

### What testing has been done?

Testing on my dev VM includes:

- Run `Clone Template Main.vi` to clone the source.
- Confirm Cancel works - deletes clone and Public Documents files. 
- Confirm the CD shows in System Explorer under Custom Devices>National Instruments and appears to function.
